.github/workflows/sycl-linux-run-tests.yml

```diff
@@ -299,6 +298,17 @@ jobs:
         rm -f ${{ inputs.toolchain_artifact_filename }}
         echo PATH=$PWD/toolchain/bin/:$PATH >> $GITHUB_ENV
         echo LD_LIBRARY_PATH=$PWD/toolchain/lib/:$LD_LIBRARY_PATH >> $GITHUB_ENV
+    - name: Download SYCL toolchain using release tag
```

Comment:
Should this step and the previous be mutually exclusive? What does it mean if both `Extract/Setup SYCL toolchain` and `Download SYCL toolchain using release tag` run?

Reply:
Well, they kind of are exclusive. Artifact is only set for `workflow_call` trigger and release tag for `workflow_dispatch`.

Reply:
can we set up the toolchain in a single step with an if statement to check which variable is set?

Reply:
ok yeah i see the problem, combining the three cases (using build step artifact from workflow call, using build step artifact from workflow dispatch, using nightly for workflow dispatch) would be pretty annoying, i guess some of it is our fault for trying to support a ton of cases but github definitely does not make it easy